src/Plugin/AddPathPlugin.php

```diff
@@ -32,17 +41,39 @@ public function __construct(UriInterface $uri)
         }
 
         $this->uri = $uri;
+
+        $resolver = new OptionsResolver();
+        $this->configureOptions($resolver);
+        $options = $resolver->resolve($config);
+
+        $this->smart = $options['smart'];
     }
 
     /**
      * {@inheritdoc}
      */
     public function handleRequest(RequestInterface $request, callable $next, callable $first)
     {
-        $request = $request->withUri($request->getUri()
-            ->withPath($this->uri->getPath().$request->getUri()->getPath())
-        );
+        $prepend = $this->uri->getPath();
+        $path = $request->getUri()->getPath();
+
+        if (!$this->smart || substr($path, 0, strlen($prepend)) !== $prepend) {
```

Comment:
To get back to my use case, it's about using `Link` header href attributes. As of https://tools.ietf.org/html/rfc5988 §5.1 this IRI can be relative so I'm not sure that in a future release gitlab will send relative IRIs.

I did this way to be more consistant with `AddHostPlugin` which owns the replace logic.